### PR TITLE
Sample: add two compatiable interfaces for old code.

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?
Sample: add two compatiable interfaces for old code.

## How was this patch tested?
unit tests

## Related links or issues (optional)
Meet some new comments in #781

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala [hunk 1]
```diff
@@ -35,7 +36,7 @@ object toAutoencoderBatch {
 class toAutoencoderBatch[T] extends Transformer[MiniBatch[T], MiniBatch[T]] {
   override def apply(prev: Iterator[MiniBatch[T]]): Iterator[MiniBatch[T]] = {
     prev.map(batch => {
-      MiniBatch(batch.data, batch.data)
+      MiniBatch(batch.getInput().asInstanceOf[Tensor[T]], batch.getInput().asInstanceOf[Tensor[T]])
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala]: toTensor?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala [hunk 2]
```diff
@@ -69,7 +69,7 @@ object Test {
 
       val flow = rdd.mapPartitions(iter => {
         iter.map(batch => {
-          var curInput = batch.data
+          var curInput = batch.getInput().asInstanceOf[Tensor[Float]]
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala]: toTensor?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala (hunk 1 above)
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala]: Changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala (hunk 2 above)
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala]: Changed

**Comment by qiuxin2012** [comment]
jenkins passed.

**Pull request merged**